plugins/modules/features.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: features
+
+short_description: Query OpenShift features
+
+version_added: "1.0.0"
+
+description: Query supported features for a given OpenShift version
+
+options:
+  openshift_version:
+    description: Version of OpenShift
+    required: True
+    type: str
+
+  cpu_architecture:
+    description: The CPU architecture of the OpenShift version
+    required: False
+    choices: [x86_64, aarch64, arm64, ppc64le, s390x, multi]
+    default: x86_64
+    type: str
+
+  platform_type:
+    description: The provider platform type
+    required: False
+    choices: [baremetal, none, nutanix, vsphere, external]
+    type: str
+
+  external_platform_name:
+    description: External platform name when platform_type is external
+    required: False
+    type: str
+
+author:
+    - Chris Wheeler (@clwheel)
+"""
+
+EXAMPLES = r"""
+- name: Query OpenShift features
+  features:
+    openshift_version: 4.16.19
+    cpu_architecture: x86_64
+    platform_type: baremetal
+"""
+
+RETURN = r"""
+features:
+  description: A list of supported OpenShift features
+  type: dict
+  returned: always
+  sample: { "CLUSTER_MANAGED_NETWORKING": "supported",
```

Comment:
I saw the below as output when I curled, looks like the response is a dict mapped to the `features` key, should the same be updated to reflect that? CC: @clwheel @tonyskapunk 
```
[openshift@openshift assistedinstaller]$ curl --location https://api.openshift.com/api/assisted-install/v2/support-levels/features?openshift_version=4.16 --header "Authorization: Bearer ${token}" 
{"features":{"BAREMETAL_PLATFORM":"supported","CLUSTER_MANAGED_NETWORKING":"supported","CNV":"supported","CUSTOM_MANIFEST":"supported","DUAL_STACK":"supported","DUAL_STACK_VIPS":"supported","EXTERNAL_PLATFORM":"supported","EXTERNAL_PLATFORM_OCI":"supported","FULL_ISO":"supported","LSO":"supported","LVM":"supported","MCE":"supported","MINIMAL_ISO":"supported","NONE_PLATFORM":"supported","NUTANIX_INTEGRATION":"supported","ODF":"supported","OVN_NETWORK_TYPE":"supported","SDN_NETWORK_TYPE":"unavailable","SINGLE_NODE_EXPANSION":"supported","SKIP_MCO_REBOOT":"supported","SNO":"supported","USER_MANAGED_NETWORKING":"supported","VIP_AUTO_ALLOC":"unavailable","VSPHERE_INTEGRATION":"supported"}}

```

Reply:
@vjayaramrh If you look at [line 55](https://github.com/vjayaramrh/assistedinstaller/pull/24/files/8240ef2b0f708f38dca7aa5b85d3629f7819cc8c#diff-bc610870e170ba96b364b2f7c7fc668fb3e1e43d96b4c87eb608fd04f1f3a628R55) it shows that `features` is the key of the dictionary that is returned.

Reply:
Thanks @clwheel